XenAdmin/Dialogs/PropertiesDialog.cs

```diff
@@ -338,8 +338,24 @@ private void btnOK_Click(object sender, EventArgs e)
 
             // Add a save changes on the beginning of the actions to enact the alterations that were just changes to the xenObjectCopy.
             // Must come first because some pages' SaveChanges() rely on modifying the object via the xenObjectCopy before their actions are run.
-            actions.Insert(0, new SaveChangesAction(xenObjectCopy, xenObjectBefore, true));
 
+            if(xenObjectBefore is VMSS)
+            {
+                XenAPI.VMSS VMSSObj = xenObjectBefore as XenAPI.VMSS;
+                if (VMSSObj.policy_type == policy_backup_type.snapshot_with_quiesce)
+                {
+                    actions.Insert(0, new SaveChangesAction(xenObjectCopy, xenObjectBefore, true));
+                }
+                else
+                {
+                    actions.Insert(actions.Count, new SaveChangesAction(xenObjectCopy, xenObjectBefore, true));
+                }
+            }
+            else /* VMPP object*/

```

Comment:
I suggest to remove the comment /* VMPP object*/ from here, because on this path xenObjectBefore can be any XenObject type other than VMSS.